sdk/openai/azure-ai-openai-assistants/src/test/java/com/azure/ai/openai/assistants/AzureRunThreadSyncTest.java

```diff
@@ -136,7 +173,9 @@ public void createThreadAndRun(HttpClient httpClient, OpenAIServiceVersion servi
             assertTrue(openAIPageableListOfThreadMessage.getData().size() > 1);
 
             // Delete the created thread
-            client.deleteThread(threadId);
-        }, mathTutorAssistant.getId());
+            deleteThread(client, threadId, LOGGER);
+        }, mathTutorAssistantId);
+        // Delete the created assistant
+        deleteMathTutorAssistant(client, mathTutorAssistantId, LOGGER);
```

Comment:
I am starting to think how we can be smarter about cleanups. There is quite a bit of teardown that we need to do.

Reply:
Yeah It is quite a bit of teardown happened. We need to think about a better way to reduce the cost.